src/game/Strategic/Strategic_Movement.cc

```diff
@@ -2720,8 +2720,13 @@ void RetreatGroupToPreviousSector(GROUP& g)
 {
 	AssertMsg(!g.fBetweenSectors, "Can't retreat a group when between sectors!");
 
-	UINT8 direction = 255;
-	if (g.ubPrev.x != 16 || g.ubPrev.y != 16)
+	if (g.ubPrev.x == 16 && g.ubPrev.y == 16)
+	{
+		// Group doesn't have a previous sector. Create one first.
+		CalculateGroupRetreatSector(&g);
+	}
+
+	UINT8 direction;
 	{ // Group has a previous sector
```

Comment:
You can remove this level of braces and unindent.

Reply:
I didn't want to touch more lines than necessary for the sake of git blame, and I liked the fact that these braces limit the scope of delta, although C++17 would allow me to move the initialization into the if() statement instead. I'll change it if you want.

Reply:
Can be a separate commit, but I'll also say I definitely don't like declarations in if statements themselves.